### ISSUE TITLE
OpenSSL::SSL::SSLError for non-SSL URL

### ISSUE THREAD

**Issue by rocketnova** [sdsykes/fastimage#53]
I ran into a really odd problem where I was getting the following error when trying to get the size of this image: http://thechangelog.com/wp-content/uploads/warp.gif

The error was:
````
~/.rbenv/versions/1.9.3-p448/lib/ruby/1.9.1/net/http.rb:800:in `connect': SSL_connect SYSCALL returned=5 errno=0 state=SSLv2/v3 read server hello A (OpenSSL::SSL::SSLError)
````

I tried running the test from #23, first comment by sdsykes, and I got the same result where `a.port` returns 80 and `b.port` returns nil.

A stack trace shows that the error comes from:
````
~/.rbenv/versions/1.9.3-p448/lib/ruby/gems/1.9.1/gems/fastimage-1.7.0/lib/fastimage.rb:218:in `fetch_using_http_from_parsed_uri'`
````

I eventually just stopped trying to retrieve the size from that image, but thought you should know.

**Comment by sdsykes** [comment]
Thanks for this report. http://thechangelog.com/wp-content/uploads/warp.gif redirects to https://changelog.com/wp-content/uploads/warp.gif, so that's why SSL is being used.

I'll paraphrase from here http://stackoverflow.com/questions/25814210/opensslsslsslerror-ssl-connect-syscall-returned-5-errno-0-state-sslv3-read : 

This is a problem at the server site. It looks like the server is exclusively accepting TLS 1.2 and does not show the usual behavior when the client requests something lesser (like downgrading or sending SSL alert) but instead just closes the connection.

TLS 1.2 is not supported by OpenSSL 0.9.8. You get TLS 1.2 only when upgrading to OpenSSL 1.0.1.

In my version of Ruby 1.9.3 OpenSSL::OPENSSL_VERSION reports "OpenSSL 0.9.8y 5 Feb 2013", and like you I can't fetch this file. In Ruby 2.2.0 I have "OpenSSL 1.0.1k 8 Jan 2015", and I can fetch the file ok.

I tried setting different ciphers and SSL/TLS versions, but without any success.

Basically the version of Ruby/OpenSSL you are using can't talk properly to the server with that image on. I don't think I can fix anything in FastImage for this, so I'm going to close it, with the note that upgrading Ruby would be a a solution if it's possible for you.

**Comment by rocketnova** [comment]
Thanks for looking into this. Like you, I researched OpenSSL 0.9.8 vs OpenSSL 1.0.1, but the critical piece I missed was that the URL redirected to the https version. Thanks!
